### ISSUE TITLE
Possible memory leak

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/area-model-algebra#9]
For https://github.com/phetsims/qa/issues/1031. This memory test was done on Win 11 Chrome, with snapshots taken one minute apart. The snapshots are increasing and do not seem to level off. While the leak doesn't seem large, it does seem to be there.
![memleak](https://github.com/phetsims/area-model-algebra/assets/41024075/de160057-e13d-4c5f-b7e8-0bc154962f9e)

**Comment by KatieWoe** [comment]
This is also in https://github.com/phetsims/qa/issues/1032
![memfail](https://github.com/phetsims/area-model-algebra/assets/41024075/c9d76d87-57fc-4835-a4f9-de5b3a445d4a)

**Comment by Luisav1** [comment]
Met with @jonathanolson to trace the memory leak which was due to `PatternStringProperty`'s and a child in the `lineNode` in `CalculationLinesNode.js` not being disposed of properly.

No other memory leaks seem present as this is now the snapshots of AMA:
![image](https://github.com/phetsims/area-model-algebra/assets/52978048/bc8cf76a-0706-46ec-8d11-b925eeff9058)

And AMM:
![image](https://github.com/phetsims/area-model-algebra/assets/52978048/02e8e2da-3f3c-44ce-9743-0ced8f389304)

**Comment by Luisav1** [comment]
Closing since there's no more work here.

**Comment by KatieWoe** [comment]
Also the case in https://github.com/phetsims/qa/issues/1041
![memagain](https://github.com/phetsims/area-model-algebra/assets/41024075/89a5d366-25c1-4017-9737-616371dcfa2e)

**Comment by zepumph** [comment]
I can take a look on Wednesday. Does that timeline work?

**Comment by Luisav1** [comment]
Thanks @zepumph, that timeline works great!

**Comment by zepumph** [comment]
Looks like this was done in https://github.com/phetsims/scenery/issues/1592#issuecomment-1954543059. @Luisav1 anything else here?

**Comment by zepumph** [comment]
After discussing with @jonathanolson, I'll do another memory test and a bit more investigation before confirming that this is fixed

**Comment by zepumph** [comment]
I'm still seeing a SoundClip memory leak here, 1870 added clips in ~10 minutes. I'll take a look.

**Comment by zepumph** [comment]
Ahhh, the AudioContext hadn't been started. As soon as I clicked in the sim. The memory dropped from 116 to 102. And 2300 SoundClips were disposed. I believe there is no more memory leak. Thanks all!

**Comment by zepumph** [comment]
How long were you fuzzing to get the 30MB increase?

**Comment by KatieWoe** [comment]
I take a snapshot at the start, then one every minute, so 10 minutes.

**Comment by Luisav1** [comment]
I got help from @marlitas and @jbphet with this.

These were the results @marlitas got from a memory test on main (since this couldn't be done on rc.2 due to https://github.com/phetsims/area-model-algebra/issues/14), which we thought looked good.
![image](https://github.com/phetsims/area-model-algebra/assets/52978048/edcc0b72-0f2f-424d-a7b7-61ca1f04d816)

@jbphet and I also looked into it with a more definitive test where the screens are reset in between. We looked into all the objects created and nothing stood out in AMA. We saw a few `partition` nodes but concluded those seem to be leftover from the challenges.
For AMM, we saw similar results with `partition` nodes. However, AMM did have some of these `partition` nodes stemming from the `AnimatedPanZoomListener` that was added in #12. Looking into similar implementations, it seems no unlinking needed to be done so this is also normal behaviour.
So maybe a very slow memory leak, if at all, might be there but that would only appear after hours of use, so we've concluded that the sims are good to go.
